util.go

```diff
@@ -16,36 +17,92 @@ const (
+// even FATAL for production environments, as logs are sent to logjam and/or Graylog
+// anyway.
+type Logger struct {
+	*log.Logger          // The embedded log.Logger.
+	LogLevel    LogLevel // Log attemtps with a log level lower than this field are not forwarded to the embbeded logger.
+}
+
+func (l *Logger) logf(ctx context.Context, severity LogLevel, format string, args ...interface{}) {
+	line := fmt.Sprintf(format, args...)
 	if request := GetRequest(ctx); request != nil {
-		request.Log(severity, fmt.Sprintf(format, args...))
+		request.Log(severity, line)
+	}
+	if severity >= l.LogLevel {
+		l.Output(3, line)
 	}
 }
 
-// LogDebug calls Log with DEBUG severity.
-func LogDebug(ctx context.Context, format string, args ...interface{}) {
-	Log(ctx, DEBUG, format, args...)
+func (l *Logger) log(ctx context.Context, severity LogLevel, args ...interface{}) {
+	line := fmt.Sprint(args...)
+	if request := GetRequest(ctx); request != nil {
+		request.Log(severity, line)
+	}
+	if severity >= l.LogLevel {
+		l.Output(3, line)
+	}
+}
+
+// Debugf logs with DEBUG severity.
+func (l *Logger) Debugf(ctx context.Context, format string, args ...interface{}) {
+	l.logf(ctx, DEBUG, format, args...)
+}
+
+// Debug logs with DEBUG severity.
+func (l *Logger) Debug(ctx context.Context, args ...interface{}) {
+	l.log(ctx, DEBUG, args...)
+}
+
+// Infof logs with INFO severity.
+func (l *Logger) Infof(ctx context.Context, format string, args ...interface{}) {
+	l.logf(ctx, INFO, format, args...)
+}
+
+// Info logs with INFO severity.
+func (l *Logger) Info(ctx context.Context, args ...interface{}) {
+	l.log(ctx, INFO, args...)
+}
+
+// Warnf logs WARN severity.
+func (l *Logger) Warnf(ctx context.Context, format string, args ...interface{}) {
+	l.logf(ctx, WARN, format, args...)
+}
+
+// Warn logs with WARN severity.
+func (l *Logger) Warn(ctx context.Context, args ...interface{}) {
+	l.log(ctx, WARN, args...)
 }
 
-// LogInfo calls Log with INFO severity.
-func LogInfo(ctx context.Context, format string, args ...interface{}) {
-	Log(ctx, INFO, format, args...)
+// Errorf logs with ERROR severity.
+func (l *Logger) Errorf(ctx context.Context, format string, args ...interface{}) {
+	l.logf(ctx, ERROR, format, args...)
 }
 
-// LogWarn calls Log with WARN severity.
-func LogWarn(ctx context.Context, format string, args ...interface{}) {
-	Log(ctx, WARN, format, args...)
+// Error logs with ERROR severity.
+func (l *Logger) Error(ctx context.Context, args ...interface{}) {
+	l.log(ctx, ERROR, args...)
 }
 
-// LogError calls Log with ERROR severity.
-func LogError(ctx context.Context, format string, args ...interface{}) {
-	Log(ctx, ERROR, format, args...)
+// FatalPanicf logs with FATAL severity, then panics. Please note that the standard Logger
+// method exits the program, which is usually not a desired outcome in a long running
+// server application.
+func (l *Logger) FatalPanicf(ctx context.Context, format string, args ...interface{}) {
```

Comment:
I don't like the `FatalPanic` methods and would say we don't need them. `Fatal` in my eyes is an unrecoverable error that should bubble up and be catched by a `recover` or terminate the program. If I see a `FATAL` logline in logjam I assume that my request failed and the error was caught and logged by logjam.

Reply:
I would like to have a method that can log at log level fatal, but this is already taken by the standard logger methods. And the Fatal method of the standard logger calls os.Exit, so nothing bubbles up. We could make it more explicit what happens by renaming it to LogFatalAndPanicf, but I don't want to drop it.

Reply:
fair enough, let's keep it then.

Reply:
So, rename yes or no?

Reply:
no, let's keep it like it is.